src/stream/src/executor/dispatch.rs

```diff
@@ -956,7 +975,13 @@ impl Dispatcher for SimpleDispatcher {
             .exactly_one()
             .expect("expect exactly one output");
 
-        let chunk = chunk.project(&self.output_indices);
+        let chunk = if self.output_indices.len() < chunk.columns().len() {
+            chunk
+                .project(&self.output_indices)
+                .eliminate_adjacent_noop_update()
```

Comment:
Is it possible to have no-op update even without output indices?

Reply:
We can, but I am afraid it is abused (e.g. do it for every executor), so maybe just in those places with column pruning.

Reply:
I see. So actually you're meant to do elimination after `Materialize`.